pkg/operator/controller/ingress/controller.go

```diff
@@ -468,6 +468,11 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 			effectiveStrategy.LoadBalancer.DNSManagementPolicy = operatorv1.UnmanagedLoadBalancerDNS
 		}
 
+		// OpenStack platform does not support managed DNS for load balancers.
+		if platformStatus.Type == configv1.OpenStackPlatformType && platformStatus.OpenStack != nil {
```

Comment:
Do we need to check `platformStatus.OpenStack != nil` here? I think it's safer to just check for the Type since we aren't using any fields below it, in the rare scenario that there aren't any `status.openStack` values set.

```suggestion
if platformStatus.Type == configv1.OpenStackPlatformType {
```